src/ucred.rs

```diff
@@ -0,0 +1,76 @@
+use libc::{uid_t, gid_t};
+
+/// Credentials of a process
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub struct UCred {
+    /// UID (user ID) of the process
+    pub uid: uid_t,
+    /// GID (group ID) of the process
+    pub gid: gid_t,
+}
+
+#[cfg(target_os = "linux")]
+pub use self::impl_linux::get_peer_cred;
+
+#[cfg(target_os = "macos")]
+pub use self::impl_macos::get_peer_cred;
+
+#[cfg(target_os = "linux")]
+pub mod impl_linux {
+    use libc::{getsockopt, SOL_SOCKET, SO_PEERCRED, c_void};
+    use std::{io, mem};
+    use UnixStream;
+    use std::os::unix::io::AsRawFd;
+
+    use libc::ucred as UCred;
```

Comment:
Can this avoid being renamed? It confused me when considering the outer one as well.